### ISSUE TITLE
[enhancement]: upload model via model manager

### ISSUE THREAD

**Issue by sammcj** [invoke-ai/InvokeAI#6179]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What should this feature add?

It would be great to have the ability to upload a model from your local machine via the web interface.

This would be useful for sites such as Civit which often only allow Model/LoRA downloads when logged in and that don't work well with / don't have an API key available in InvokeAI's environment.

Often I download a LoRA or Model locally, maybe try it out with something on my machine quickly, then I have to rsync it to my server running InvokeAI, chmod/chown, then find the path in the mounted container volume, copy it, load InvokeAI, go to the model manager -> add model -> from path -> paste the path in relative to the containers fs -> click search -> find the model -> import it etc... which is a right pain in the bum.

Ideally there'd be either or both:

1. An "upload model" button that pops up a browser file dialogue to select a safetensors file to upload.
2. The ability to drag a safetensors model onto the invokeAI web interface and have it upload.

**Comment by psychedelicious** [comment]
I don't think we want to implement a separate frontend for uploading models in the UI.

You can set an API key, though: https://invoke-ai.github.io/InvokeAI/features/CONFIGURATION/#model-marketplace-api-keys

Does that address your use-case?

**Comment by sammcj** [comment]
I was thinking more just another tab or button in or next to the scan folder item:

<img width="1084" alt="image" src="https://github.com/invoke-ai/InvokeAI/assets/862951/3ad1c4aa-a78f-4fc7-b2e6-0ae27c83edae">

or simply if you dragged a model file into the model manager pane, similar to how you do with importing images.

I looked at adding API keys via the config file - but as it's committed to git I didn't really want to add secrets there. If InvokeAI can read the secrets from other files or environment variables that may be good enough as a workaround though?

**Comment by psychedelicious** [comment]
Yes, you can read from environment variables. There's an example here: https://invoke-ai.github.io/InvokeAI/features/CONFIGURATION/#environment-variables

Uploading files would similarly require the API key, no?

To clarify, handling for uploading files to whichever model marketplace, via API or UI, is out of scope. Big can of worms - we don't want to be responsible for all of that complexity.

**Comment by sammcj** [comment]
Oh I missed the env vars for remote tokens! Sorry about that.

> Uploading files would similarly require the API key, no?
> To clarify, handling for uploading files to whichever model marketplace, via API or UI, is out of scope. Big can of worms - we don't want to be responsible for all of that complexity.

Sorry no, I meant from the users local machine to InvokeAI via the browser

**Comment by psychedelicious** [comment]
Ohhhh - I misunderstood. That's much more approachable and a nice UX improvement. I think we'd be happy to review a PR to do this. @hipsterusername thoughts (see message just above mine)?

**Comment by sammcj** [comment]
Phew, thought maybe my coffee deprived brain was spewing nonsense 😅 

Something like this in the model manager:

<img width="2189" alt="image" src="https://github.com/invoke-ai/InvokeAI/assets/862951/bd6508ad-ef8a-4b0f-8641-06f56e8261b2">

**Comment by magnusviri** [comment]
I would love this. I also use Invoke on a remote computer and have to copy the model files to the server before I can upload them. I'm starting to learn how to train my own models and I don't want to have to post them to another site in order to upload them.

**Comment by hipsterusername** [comment]
This would be, as the kids say, “dope as hell”.

By all means, give it a whirl!
